packages/phone-number-privacy/src/database/wrappers/account.ts

```diff
@@ -10,42 +11,47 @@ function accounts() {
  * Returns how many queries the account has already performed.
  */
 export async function getPerformedQueryCount(account: string): Promise<number> {
+  logger.debug('Getting performed query count')
   try {
     const queryCounts = await accounts()
       .where(ACCOUNTS_COLUMNS.address, account)
       .select(ACCOUNTS_COLUMNS.numLookups)
       .first()
     return queryCounts === undefined ? 0 : queryCounts[ACCOUNTS_COLUMNS.numLookups]
   } catch (e) {
-    console.error(ErrorMessages.DATABASE_GET_FAILURE, e)
+    logger.error(ErrorMessages.DATABASE_GET_FAILURE, e)
     return 0
   }
 }
 
+async function getAccountExists(account: string): Promise<boolean> {
+  const existingAccountRecord = await accounts()
+    .where(ACCOUNTS_COLUMNS.address, account)
+    .first()
+  return !!existingAccountRecord
+}
+
 /*
  * Increments query count in database.  If record doesn't exist, create one.
  */
-export async function incrementQueryCount(account: string): Promise<void | number> {
-  await ((await incrementExistingQueryCount(account)) || (await addNewQueryCount(account)))
-}
-
-async function incrementExistingQueryCount(account: string) {
+export async function incrementQueryCount(account: string) {
+  logger.debug('Incrementing query count')
   try {
-    return await accounts()
-      .where(ACCOUNTS_COLUMNS.address, account)
-      .increment(ACCOUNTS_COLUMNS.numLookups, 1)
+    if (await getAccountExists(account)) {
```

Comment:
What do you think about using upsert here?

Reply:
Agreed, we'll do that in a follow-up